services/src/main/java/org/keycloak/protocol/ciba/resolvers/DefaultCIBALoginUserResolver.java

```diff
@@ -1,32 +1,72 @@
 package org.keycloak.protocol.ciba.resolvers;
 
+import org.apache.commons.lang.StringUtils;
+import org.jboss.logging.Logger;
+import org.keycloak.jose.jws.JWSInput;
+import org.keycloak.jose.jws.JWSInputException;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.UserModel;
 import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.ciba.CIBAErrorCodes;
+import org.keycloak.representations.IDToken;
+import org.keycloak.representations.LoginHintToken;
+import org.keycloak.services.ErrorResponseException;
+
+import javax.ws.rs.core.Response;
 
 public class DefaultCIBALoginUserResolver implements CIBALoginUserResolver {
 
-    private KeycloakSession session;
+    private static final Logger logger = Logger.getLogger(DefaultCIBALoginUserResolver.class);
+
+    private final KeycloakSession session;
 
     public DefaultCIBALoginUserResolver(KeycloakSession session) {
         this.session = session;
     }
 
     @Override
     public UserModel getUserFromLoginHint(String loginHint) {
-        return KeycloakModelUtils.findUserByNameOrEmail(session, session.getContext().getRealm(), loginHint);
+        UserModel userModel = KeycloakModelUtils.findUserByNameOrEmail(session, session.getContext().getRealm(), loginHint);
+        if (userModel == null) {
+            throw new ErrorResponseException(CIBAErrorCodes.UNKNOWN_USER_ID, "no user found", Response.Status.BAD_REQUEST);
+        }
+        return userModel;
     }
 
     @Override
     public UserModel getUserFromLoginHintToken(String loginHintToken) {
-        // not yet supported
-        return null;
+        try {
+            JWSInput input = new JWSInput(loginHintToken);
+            LoginHintToken token = input.readJsonContent(LoginHintToken.class);
+
+            String username = StringUtils.isNotBlank(token.getPreferredUsername()) ? token.getPreferredUsername() : token.getEmail();
+            UserModel userModel = KeycloakModelUtils.findUserByNameOrEmail(session, session.getContext().getRealm(), username);
+            if (userModel == null) {
+                throw new ErrorResponseException(CIBAErrorCodes.UNKNOWN_USER_ID, "no user found", Response.Status.BAD_REQUEST);
+            }
+            return userModel;
+        } catch (JWSInputException e) {
+            logger.warn("Failed verify user hint token", e);
+            throw new ErrorResponseException(CIBAErrorCodes.INVALID_REQUEST, "token invalid", Response.Status.BAD_REQUEST);
+        }
     }
 
     @Override
-    public UserModel getUserFromIdTokenHint(String idToken) {
-        // not yet supported
-        return null;
+    public UserModel getUserFromIdTokenHint(String idTokenHint) {
+        try {
+            JWSInput input = new JWSInput(idTokenHint);
+            IDToken token = input.readJsonContent(IDToken.class);
```

Comment:
It might be better to verify the following points.

- signature verification
It might be helpful to refer to `AccessTokenIntrospectionProvider.verifyAccessToken` for token signature verification.

- issuer verification
To check whether this id token is issued by this keycloak.
"iss" field specifies Issuer of the id token.
 
- audience verification
To check whether the sender of this id token is the audience of this token.
"aud" field specifies the audience of this id token.